rules/coding-style/src/Rector/Throw_/AnnotateThrowablesRector.php

```diff
@@ -173,83 +211,210 @@ private function identifyThrownThrowablesInStaticCall(StaticCall $staticCall): a
         if (! $thrownClass instanceof FullyQualified) {
             throw new ShouldNotHappenException();
         }
-        $classFqn = implode('\\', $thrownClass->parts);
-        $methodNode = $thrownClass->getAttribute('nextNode');
-        $methodName = $methodNode->name;
 
-        return $this->extractMethodReturnsFromDocblock($classFqn, $methodName);
+        return $this->extractMethodReturns($thrownClass, $thrownClass->getAttribute('nextNode'));
     }
 
-    private function extractMethodReturnsFromDocblock(string $classFqn, string $methodName): array
+    private function extractMethodReturns(FullyQualified $fullyQualified, $methodNode): array
     {
+        $classFqn = implode('\\', $fullyQualified->parts);
+        $methodName = $methodNode->name;
         $reflectedMethod = new ReflectionMethod($classFqn, $methodName);
         $methodDocblock = $reflectedMethod->getDocComment();
 
-        // copied from https://github.com/nette/di/blob/d1c0598fdecef6d3b01e2ace5f2c30214b3108e6/src/DI/Autowiring.php#L215
-        $result = Strings::match((string) $methodDocblock, self::RETURN_DOCBLOCK_TAG_REGEX);
-        if ($result === null) {
+        if (! is_string($methodDocblock)) {
             return [];
         }
 
-        $returnTags = explode('|', str_replace('@return ', '', $result[0]));
+        $returnTags = $this->extractTagFromStringedDocblock($methodDocblock, '@return');
+
         $returnClasses = [];
         foreach ($returnTags as $returnTag) {
             $returnClasses[] = Reflection::expandClassName($returnTag, $reflectedMethod->getDeclaringClass());
         }
 
-        $this->foundThrownClasses = $returnClasses;
-
         return $returnClasses;
     }
 
-    private function annotateThrowable(Throw_ $node): void
+    private function extractFunctionAnnotatedThrows(FullyQualified $fullyQualified): array
```

Comment:
This method could be extracted to standalone service,
e.g. FunctionThrowsAnnotationResolver

Reply:
Yes, and it isn’t the only one 😅

Which namespace should have I to use?

Reply:
Maybe `AnnotationReflection`

Reply:
In which folder do I put the file?

Reply:
Same answer https://github.com/rectorphp/rector/pull/2998#discussion_r388158583